Add release notes for 0.3

geeknoid opened the pull request:
This is an initial pass on the 0.3 release notes. I generated those by going through all the commits since 0.2 and surfacing anything more than a bug fix or build-related change. There isn't much there. Please let me know if you would like to surface anything user-oriented for your area.

Staging site: https://geeknoid.github.io/istio.github.io/

ldemailly commented on _docs/reference/notes/0.2.md:
```diff
@@ -0,0 +1,91 @@
+---
+title: Istio 0.2
```
is this pretty much just moved from release-notes.md or there are other changes ?

might be useful to start directly with the version and just keep adding new files overtime rather than create them later

ldemailly commented on _docs/reference/release-notes.md:
```diff
@@ -8,91 +8,34 @@ layout: docs
-- **User may get periodical 404 when accessing the application**:  We have noticed that Envoy doesn't get routes properly occasionally thus a 404 is returned to the user.  We are actively working on this [issue](https://github.com/istio/istio/issues/1038).
-- **Istio Ingress or Egress reports ready before Pilot is actually ready**: You can check the istio-ingress and istio-egress pods status in the `istio-system` namespace and wait a few seconds after all the Istio pods reach ready status.  We are actively working on this [issue](https://github.com/istio/istio/pull/1055).
-- **A service with Istio Auth enabled can't communicate with a service without Istio**: This limitation will be removed in the near future.
+- **Selective Auth**. You can now control authentication on a per-service basis via service annotations. [TBD: DOC LINK?]
```
that's one of the major features in this release
https://istio.io/faq/#collapse13
(the link doesn't actually work, but... click "security" then click "Can I enable istio auth with some services...")

cc @wattli

geeknoid replied:
Can we get a real documentation link? The FAQ can't be the only place we explain how to do this...

wenchenglu replied:
@diemtvu to help provide the doc link to per-service auth enablement

rshriram commented on _docs/reference/release-notes.md:
```diff
@@ -8,91 +8,34 @@ layout: docs
 
-- **Caching Client**. The Mixer client library used by Envoy now provides caching for Check calls and batching for Report calls, considerably reducing 
-end-to-end overhead.
+- **Secured Components**. Mixer and Pilot are now secured with mutual TLS, just like all services in a mesh.
```
Secure Control Plane Communication

geeknoid commented:
Addressed feedback.

I reorganized this so we have a cleaner/simpler way to deal with versions over time, as suggested by Laurent. There is also now a top-level redirect, so istio.io/notes gets you to where you want to go.

We just need a good link to the selective auth feature.

As before: https://geeknoid.github.io/istio.github.io

PTAL.

ldemailly commented on _docs/reference/notes/0.3.md:
```diff
@@ -0,0 +1,44 @@
+
+## Networking
+
+- **Egress RUles for TCP
```
typo ^ maybe you meant:
- **Egress Rules for TCP**

geeknoid replied:
Whoops, I hadn't finished typing the whole thing in. Sorry about that.

Fixed.

ldemailly commented on _docs/reference/notes/index.md:
```diff
@@ -1,11 +1,13 @@
 type: markdown
+redirect_from:
+  - "/docs/reference/release-notes.html"
+  - "/notes"
```
maybe `/release-notes` is enough, "/notes" alone is a bit ambiguous ?

geeknoid replied:
Done

ldemailly commented on _docs/reference/notes/index.md:
```diff
@@ -1,11 +1,13 @@
+overview: What's been happening with Istio.
 
-order: 600
+order: 10
```
this puts it after API, it should probably be second below Feature Status (or even first)

![screen shot 2017-12-05 at 11 49 05 am](https://user-images.githubusercontent.com/3664595/33627420-51aac956-d9b2-11e7-8ebf-68702d54ba82.png)

geeknoid replied:
I moved it above API. At the moment, "directories" always come after plain files, so that's as high as I can make it.
